### DISCUSSION TITLE
How to authenticate against Artifactory

### DISCUSSION THREAD

**Discussion by steffakasid** [goreleaser/goreleaser#2878] [Q&A] [answered]
Hi,

I've a question on how to authenticate against Artifactory. I've setup a Artifactory release with `name: production`. Based on this I set the following Environment Variables: `ARTIFACTORY_PRODUCTION_SECRET` and `ARTIFACTORY_PRODUCTION_USERNAME`. Anyway I always receive the following error: `401 [{Status:401 Message:Bad authentication Key}]`. The credentials should be correct as I can do a manual update with them to the same repo.

Any Idea what could be wrong?

Regards
Steffen

**Comment by caarlos0** [comment, marked as answer]
the auth is made as explained in https://goreleaser.com/customization/artifactory

under the hood, it uses the same mechanisms as the http upload pipe, so you might want to check that out too https://goreleaser.com/customization/upload/

**Reply by steffakasid** [reply]
Thanks a lot unfortunately it works now. It seems  that the Environment variables were not set properly in GitLabCI.
